source/Hazard.h

```diff
@@ -0,0 +1,57 @@
+/* Hazard.h
+Copyright (c) 2020 by Jonathan Steck
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef HAZARD_H_
+#define HAZARD_H_
+
+#include "Weapon.h"
+
+
+// Hazards are environmental effects created within systems. They are able to create
+// visual effects and damage or apply status effects to any ships within the system
+// while active.
+class Hazard : public Weapon {
+public:
+	void Load(const DataNode &node);
+	
+	const std::string &Name() const;
+	// Does the strength of this hazard deviate over time?
```

Comment:
Should this constant vs. variable strength attribute be a property of the `Hazard` itself, or of a particular system "weather" effect?

You may be able to simplify this PR by re-evaluating the responsibilities of the Hazard class and of the Weather class.

Reply:
Currently, the only thing the system controls is the probability of a weather event activating. Beyond that, all aspects of a particular hazard are handled by the `Hazard` class.